broker/lib/fabrik/UnlockResourcePoller.js

```diff
@@ -15,9 +15,12 @@ class UnlockResourcePoller {
     function poller(object, interval) {
       //TODO-PR - instead of a poller, its better to convert this to a watcher.
       const lockDetails = JSON.parse(object.spec.options);
-      return eventmesh.apiServerClient.getResource(lockDetails.lockedResourceDetails.resourceGroup, lockDetails.lockedResourceDetails.resourceType, lockDetails.lockedResourceDetails.resourceId)
-        .then((resource) => {
-          const resourceState = resource.body.status.state;
+      return eventmesh.apiServerClient.getState({
+          resourceGroup: lockDetails.lockedResourceDetails.resourceGroup,
+          resourceType: lockDetails.lockedResourceDetails.resourceType,
+          resourceId: lockDetails.lockedResourceDetails.resourceId
+        })
+        .then((resourceState) => {
```

Comment:
lockManager.unlock [method](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/data-access-layer/eventmesh/LockManager.js#L170) has retries.
For now all the locks have infinite TTL. We are thinking of keeping lock TTL to some definite time.
What should be the ideal TTL for deployment?

Reply:
Yeah found the retry logic down but did not revisit this comment.

Regarding TTL, lets not have them Infinite you will certainly run into trouble (from experience of having locks on Backup). Drive these value from configs. I would say for Create / Update / Delete / Restore operation i would go with the CF last operation timeout time (guess 7 days). And once their TTL is reached or nearing we do what even CF does since we internally Poll. i.e. fire a Delete call continously till we get a success response back. (Thats what CF does when we timeout for create requests atleast, since we will have our own pollers we must mimic the same).

For Backup operation whatever is the backup configured interval. i.e. 8hrs in AWS and 24 hrs in OpenStack.

Reply:
yes , infinite timeout is not good . We already observe issues related to long running backups etc.

Reply:
As discussed, we will change our unlock logic. For locking and unlocking we will have one state label (`locked`/`unlocked`)
Rather than deleting lock resource when we unlock, we can just change the `label` to `unlocked`.
With apiserver we achieve atomicity by `resourceVersion` which can be leveraged in all the patch calls (patch only goes through if current resourceVersion is equal to resourceVersion passed in patch metadata).